File: apps/studio/Scripts/UI/LayoutManager.cs
using System;

public class LayoutManager {
    public Vector2 Clamp(Vector2 input, Vector2 min, Vector2 max) {
        if (float.IsNaN(input.X) || float.IsInfinity(input.X)) input.X = min.X;
        if (float.IsNaN(input.Y) || float.IsInfinity(input.Y)) input.Y = min.Y;
        return new Vector2(
            Math.Clamp(input.X, min.X, max.X),
            Math.Clamp(input.Y, min.Y, max.Y)
        );
    }
}

File: apps/studio/Scripts/UI/LayoutManager.ts
export class LayoutManager {
  private readonly MIN_SIZE = { x: 200, y: 150 };
  private readonly MAX_SIZE = { x: 4096, y: 4096 };

  public Clamp(size: { x: number, y: number }): { x: number, y: number } {
    return {
      x: Math.max(this.MIN_SIZE.x, Math.min(this.MAX_SIZE.x, size.x)),
      y: Math.max(this.MIN_SIZE.y, Math.min(this.MAX_SIZE.y, size.y))
    };
  }
}